networking/mtu.md

```diff
@@ -37,65 +37,82 @@ For help using IP in IP and/or VXLAN overlays, see [Configure overlay networking
 
-- If you use VXLAN anywhere in your pod network, configure MTU size as “physical network MTU size minus 50”. 
-- If you use only IP in IP, configure MTU size as “physical network MTU size minus 20”
+- If you use WireGuard encryption configure MTU size as “physical network MTU size minus 60”.
+- If you don't use WireGuard, but use VXLAN anywhere in your pod network, configure MTU size as “physical network MTU size minus 50”.
+- If you don't use WireGuard, but use only IP in IP, configure MTU size as “physical network MTU size minus 20”
 - Set the workload endpoint MTU and the tunnel MTUs to the same value (so all paths have the same MTU)
 
+**eBPF mode**
+
+Implementation of NodePorts uses VXLAN tunnel to hand off packets from one node to another, therefore VXLAN MTU setting
+is used to set the MTUs of workloads (veths) and should be “physical network MTU size minus 50” (see above).
+
 **MTU for flannel networking**
 
-When using flannel for networking, the MTU for network interfaces should match the MTU of the flannel interface. If using flannel with VXLAN, use the “{{site.prodname}} MTU with VXLAN” column in the table above for common sizes. 
+When using flannel for networking, the MTU for network interfaces should match the MTU of the flannel interface.
+- If using flannel with WireGuard encryption, use the "{{site.prodname}} MTU with WireGuard" column in the table above for common sizes.
+- Otherwise, if using flannel with VXLAN, use the “{{site.prodname}} MTU with VXLAN” column in the table above for common sizes.
+
+#### Configure MTU
+
+> **Note**: The updated MTU used by {{site.prodname}} only applies to new workloads.
 
-#### Configure MTU for workloads
-  
-When you set the MTU, it applies to new workloads. To apply MTU changes to existing workloads, you must restart calico nodes. Restarting the calico/node pods takes values from the ConfigMap and starts rolling updates for any {{site.prodname}} tunnel network interfaces on the node. 
+Select the appropriate instructions for configuring the MTU.  This is broken down based on installation:
+-  Manifest based installation (if you are not using the quickstart guide, most non-OpenShift installs fall under this
+   category)
+-  OpenShift
+-  Operator, non-OpenShift (currently just used in the quickstart guide)
 
-Edit the `calico-config` ConfigMap to set values in FelixConfiguration. For example:
+##### **Manifest**
+
+For manifest based installations (i.e. ones that do not use the operator) edit the `calico-config` ConfigMap. For example:
 
 ```bash
 kubectl patch configmap/calico-config -n kube-system --type merge \
   -p '{"data":{"veth_mtu": "1440"}}'
 ```
 
-#### Configure MTU for overlay networking
+After updating the ConfigMap perform a rolling restart of all calico/node pods.
```

Comment:
Several questions on this:
1) Can you help educate me on why this is needed? (I thought calico/node picks up config changes without requiring a manual restart usually?)
2) How does a user actually do this rolling restart?
3) Would they also need to restart all pods on each node so they get the right MTU?

Reply:
@lxpollitt -- the proposed statement is just the rephrasing the original text. I am not sure if anything has change since the original. I will check if the restart is a must.

```
When you set the MTU, it applies to new workloads. To apply MTU changes to existing workloads, you must restart calico nodes. Restarting the calico/node pods takes values from the ConfigMap and starts rolling updates for any {{site.prodname}} tunnel network interfaces on the node. 
```

Reply:
> Can you help educate me on why this is needed? (I thought calico/node picks up config changes without requiring a manual restart usually?)

There is no code watching update to calico-config. I see some code looking for changes to _token_, _root-ca file_ but not for other configuration. This is why restart is required.

> How does a user actually do this rolling restart?
> Would they also need to restart all pods on each node so they get the right MTU?

Running command like `kubectl rollout restart daemonset calico-node -n kube-system` should restart all calico-node pods. I will add an example command in the document to make it clearer.